### PR TITLE
[DEMO-2567] Add database tutorial

### PR DESCRIPTION
**Pull request by nr-kkenney**
## Summary
Adding in a tutorial for a deployment using a database.

[NOTE]: # ( Provide a brief overview of the changes. )

## Checklist
[NOTE]: # ( Just check the ones applicable to this pull request. )
- [x] Documentation updated
- [ ] AWS Deployment configuration updated
- [ ] GCP Deployment configuration updated
- [ ] Azure Deployment configuration updated

## Related Issues
[NOTE]: # ( Provide links to any related Github issues. )

## Related Pull Requests
[NOTE]: # ( Provide links to any related pull requests. )

## Reviewers
[NOTE]: # ( Mention reviewers here! )
@aswanson-nr

### CODE REVIEW

File: tutorials/4_provision_and_instrument_a_database/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                }
+            }
+        ]
+    }
+}
+```
+
+### Note - Params
+
+Parmas is a dynamic field where any key, value pair entered is passed unto the underlying play. This is the general functionality of params across sections. Within a particular section, params may be validated & treated differently, as well as different params being needed in order for a play to successfully run.
```
Comment (polfliet) [inline tutorials/4_provision_and_instrument_a_database/README.md:120]:
```suggestion
Params is a dynamic field where any key, value pair entered is passed unto the underlying play. This is the general functionality of params across sections. Within a particular section, params may be validated & treated differently, as well as different params being needed in order for a play to successfully run.
```

**Review by polfliet — APPROVED** [review]

File: tutorials/4_provision_and_instrument_a_database/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                "id": "mysql_integration",
+                "service_ids": [
+                    "mysql"
+                ],
+                "provider": "newrelic",
+                "source_repository": "-b main https://github.com/newrelic/demo-newrelic-instrumentation.git",
+                "deploy_script_path": "deploy/databases/mysql/linux/roles",
+                "version": "1.4.0",
+                "params": {
+                    "database_port": "[service:mysql:port]"
```
Comment (aswanson-nr) [inline tutorials/4_provision_and_instrument_a_database/README.md:112]: This also needs the root password, right?

File: tutorials/4_provision_and_instrument_a_database/README.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                }
+            }
+        ]
+    }
+}
+```
+
+### Note - Params
+
```
Comment (aswanson-nr) [inline tutorials/4_provision_and_instrument_a_database/README.md:121]: Would you add a table here that goes over the different values that the database needs? I see you have the the passwords covered, but I think a similar explanation of each param would help people understand what's going on here.

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
Looks good! I just have 2 comments

File: tutorials/4_provision_and_instrument_a_database/README.md (hunk 2 above)
Comment (nr-kkenney) [reply tutorials/4_provision_and_instrument_a_database/README.md:112]: nope

Comment (nr-kkenney) [reply tutorials/4_provision_and_instrument_a_database/README.md:112]: yea this isn't required to be defined here. the instrumentation play is receiving the value ( i assume its automatically passed from the service, i remember discussing this or something similar before 🙈 , but actually forget even now where this is ).

File: tutorials/4_provision_and_instrument_a_database/README.md (hunk 3 above)
Comment (nr-kkenney) [reply tutorials/4_provision_and_instrument_a_database/README.md:131]: added. lmk if that looks okay.

File: tutorials/4_provision_and_instrument_a_database/README.md (hunk 2 above)
Comment (aswanson-nr) [reply tutorials/4_provision_and_instrument_a_database/README.md:112]: Ah yes, you're completely right!

File: tutorials/4_provision_and_instrument_a_database/README.md (hunk 3 above)
Comment (aswanson-nr) [reply tutorials/4_provision_and_instrument_a_database/README.md:131]: Looks good to me!

**Review by aswanson-nr — APPROVED** [review]
Noice!

**Pull request merged**